File: commands/moderation/createtextchannel.js
module.exports = {
    name: 'createtextchannel',
    aliases: ['ctc'],
    permissions: ["MANAGE_CHANNELS"],
    cooldown: 0,
    execute(client, message, cmd, args, Discord) {

        require('dotenv').config();
        const prefix = process.env.PREFIX

        if (cmd === 'createtextchannel') {
            const name = message.content.replace(`${prefix}createtextchannel`, '')

            message.guild.channels
                .create(name, {
                    type: 'text',
                })
                .then((channel) => {
                    const categoryId = message.channel.parentID
                    channel.setParent(categoryId)
                    message.channel.send({ embed: { color: `RANDOM`, description: `Text Channel **${name}** have been created!` } })
                })
        } else if (cmd === 'ctc') {
            const name = message.content.replace(`${prefix}ctc`, '')

            message.guild.channels
                .create(name, {
                    type: 'text',
                })
                .then((channel) => {
                    const categoryId = message.channel.parentID
                    channel.setParent(categoryId)
                    message.channel.send({ embed: { color: `RANDOM`, description: `Text Channel **${name}** have been created!` } })
                })
        }
    }
}

File: commands/Moderation/createtextchannel.py
from setting import *

@commands.has_permissions(manage_channels=True)
@client.command(aliases=["ctc"])
async def createtextchannel(ctx, *,name: str):
    
    # Basic task
    name = name.replace(" ", "-")

    # Creating channel
    await ctx.guild.create_text_channel(name, category=ctx.channel.category)
    channel = get(ctx.guild.text_channels, name=name)

    embed=discord.Embed(title="Create Text Channel", description=f"{channel.mention} channel created sucessfully", color=0x00f2ff)
    await ctx.send(embed=embed)

# Error(s)
@createtextchannel.error
async def createtextchannel_error(ctx, error):
    
    # Missing permision
    if isinstance(error, commands.MissingPermissions):
        await ctx.send(embed=discord.Embed(title="Missing Permision", description=f"Your dont have permission to create a channel, Noob", color=0x00f2ff))
        
    # Missing Agrument(s)
    elif isinstance(error, commands.MissingRequiredArgument):
        embed=discord.Embed(title="Missing Argument(s)", description="Please enter all the required agrument(s)!", color=0x00f2ff)
        embed.add_field(name="Example", value=f"{prefix}createtextchannel this is example channel")
        await ctx.send(embed=embed)
    
    # Unknown error
    else:
        await unknown_error(ctx, error)